modules/ocf/files/packages/cups/convert_failure

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+"""A script to send an email to the root mailing list upon failure of
+the convert program, when used in the raster-filter script, a
+Tea4CUPS filter.
+
+Jobs printed from one of the desktops, or from supernova or tsunami,
+first go through raster-filter, which calls convert to reduce the
+density of any PDFs, essentially rasterizing them. If convert happens
+to fail, raster-filter will pipe stderr from convert to this script,
+which takes in stdin and includes it in an email to root@ocf. Thus,
+this script should not be called casually by regular users!
+
+Based off the enforcer script for whiteout, and the
+email_ldapsync_problem script
+"""
+import os
+import socket
+import sys
+from textwrap import dedent
+
+from ocflib.misc import mail
+
+ERROR_MSG = dedent("""\
+        raster-filter encountered the following error while running convert:
+
+        {stdin}
+
+
+        tea4cups environment variables:
+        {tea_vars}
+
+        ====
+        Hostname: {hostname}
+        """.format(
+    stdin=sys.stdin.read(),
+    tea_vars='\n'.join('  {}: {}'.format(k, v)
+                       for k, v in os.environ.items() if k.startswith('TEA')),
```

Comment:
In the test emails to `root@` I've seen, this section has been empty. Do you know why that is? If it's impossible to get these environment variables, we shouldn't use them, but they can be very useful so we should try to get them working.

Reply:
I've noticed, and have been looking into this. For now, I haven't been able to get these environment variables, so I'm going to remove this section of the code, but I'm going to keep looking into this...

Reply:
Ok, the reason why these environment variables are not available is because Tea4CUPS only makes them available for prehooks and posthooks, not filters. As per the [Tea4CUPS website](http://www.pykota.com/software/tea4cups):

> To help your own commands, Tea4CUPS makes available as part of the environment several variables which can be used from the commands you use in pre and post hooks (but not filters)

Since you mentioned that these variables can be useful, I may restructure my `raster-filter` to be a Tea4CUPS prehook. Otherwise, I think what I have now should suffice...